src/js/Router.js

```diff
@@ -52,6 +56,37 @@ export default class Router extends String {
         });
     }
 
+    /**
+     * Find the first route that matches the URL
+     *
+     * @param {string} [url] The url to resolve, defaults to the current url
+     * @returns {{name: string, params: Object, query: Object, route: Route}}
+     */
+    unresolve(url) {
```

Comment:
This is also an abstraction of current() so that you can retrieve any route, params and query just from its url